### DISCUSSION TITLE
How to view scan response along with advertisement

### DISCUSSION THREAD

**Discussion by smcneill-phlsci** [hbldh/bleak#1121] [Q&A]
Using a Raspberry Pi (Raspbian) I'm able to view both the advert and the scan response using hcitools/hcidump

However, I'm unable to see the scan response when using `BleakScanner.discover(return_adv=True) `

Is this possible using Bleak?  I can't find an example and searches are not coming up with anything to help so far. If it is possible is there an example someone could point me to?

Thanks for any help!

**Comment by dlech** [comment]
The scan response is combined with advertisement data into a single data structure. It is combined at the OS level on all OSes except for Windows, so Bleak can't tell the difference and just calls everything advertisement data.

**Reply by smcneill-phlsci** [reply]
Do you mean that the data is all combined into the advertisement data so it should be viewable there? Or that the Scan Response / Advert are overwriting each other?

**Reply by dlech** [reply]
Assuming that the advertisement data and scan response don't use the same advertisement data types, they should not write over each other. Only certain advertisement data types are exposed by BlueZ. What info are you looking for?

**Reply by smcneill-phlsci** [reply]
It's a bespoke device. Scanning with the Pi manually is picking both advertisement and scan response OK. I'll double check on the advertising data types and do some more testing with Bleak and see if I can catch it. 
Thank you for you help.
